pyccl/nl_pt/ept.py

```diff
@@ -281,6 +312,24 @@ def reshape_fastpt(tupl):
         self.pk_b1 = pks[self.b1_pk_kind]
         self.pk_bk = pks[self.bk2_pk_kind]
 
+        # ak power spectrum
+        pksa = {}
+        if 'nonlinear' in [self.ak2_pk_kind]:
+            pksa['nonlinear'] = np.array([cosmo.nonlin_matter_power(
+                                          self.k_s, a)for a in self.a_s])
+        if 'linear' in [self.ak2_pk_kind]:
+            pksa['linear'] = np.array([cosmo.linear_matter_power(self.k_s, a)
+                                      for a in self.a_s])
+        if 'pt' in [self.ak2_pk_kind]:
+            if 'linear' in pksa:
+                pka = pksa['linear']
+            else:
+                pka = np.array([cosmo.linear_matter_power(self.k_s, a)
+                                for a in self.a_s])
+            pka += self._g4T*self.one_loop_dd[0]
+            pksa['pt'] = pka
+        self.pk_ak = pksa[self.ak2_pk_kind]
+
```

Comment:
Maybe I missed something, but wouldn't you achieve the same just adding `self.ak2_pk_kind` to the previous if-else statement, and then doing
```
self.pk_ak = pks[self.ak2_pk_kind]
```
?

Reply:
@jablazek @CarterDW , I have just submitted a commit that implements this the way I meant it. Sorry this was easier than explaining here with words. Check out the differences [here](https://github.com/LSSTDESC/CCL/pull/1137/commits/c551530432439df6ba00ade3159851bb0ff7be5f), and feel free to disagree with me!